### PR TITLE
[6670] Set up config for sandbox environment

### PR DESCRIPTION
**Pull request by darokel**
### Context

This PR configures a new Register environment (Sandbox) to be used for integration testing against the new Register API and DQT. 

### Changes proposed in this pull request

- Refactors post deploy GH action to parallelise deploys between `productiondata` and `sandbox` 
- Configures new environment in the application
- Terraform scripts to setup and configure the environment in Azure

### Guidance to review

Some points:

- We're trying to mirror this based off production
- We do not need the nightly db restore as we'll seed prod data one time. 
- We'll need background workers configured for this environment 
- We'll need a friendly gov domain name to be able to give to providers

Anything else missing?

### CODE REVIEW

File: terraform/aks/workspace-variables/sandbox.tfvars.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "azure_maintenance_window": {
+    "day_of_week": 0,
+    "start_hour": 2,
+    "start_minute": 0
+  },
+  "postgres_flexible_server_sku": "GP_Standard_D2ds_v4",
+  "azure_enable_backup_storage": false,
+  "deploy_temp_data_storage_account": true,
+  "azure_tempdata_storage_account_name": "s189p01registerpdttmp",
+  "azure_resource_group_name": "s189p01-rtt-pdt-rg",
```
Comment (RMcVelia) [inline terraform/aks/workspace-variables/sandbox.tfvars.json:34]: would have to be 189p01-rtt-sbx-rg

File: terraform/aks/workspace-variables/sandbox.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "azure_maintenance_window": {
+    "day_of_week": 0,
+    "start_hour": 2,
+    "start_minute": 0
+  },
+  "postgres_flexible_server_sku": "GP_Standard_D2ds_v4",
+  "azure_enable_backup_storage": false,
+  "deploy_temp_data_storage_account": true,
+  "azure_tempdata_storage_account_name": "s189p01registerpdttmp",
```
Comment (RMcVelia) [inline terraform/aks/workspace-variables/sandbox.tfvars.json:33]: would have to be s189p01registersbxtmp

File: terraform/aks/workspace-variables/sandbox.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "azure_resource_group_name": "s189p01-rtt-pdt-rg",
+  "azure_storage_account_replication_type": "RAGRS",
+  "gov_uk_host_names": [
+    "sandbox.register-trainee-teachers.service.gov.uk",
+    "sandbox.register-trainee-teachers.education.gov.uk"
+  ],
+  "enable_alerting": false,
+  "enable_monitoring": true,
```
Comment (RMcVelia) [inline terraform/aks/workspace-variables/sandbox.tfvars.json:41]: do we want monitoring for this env? We don't monitor productiondata.

Comment (darokel) [reply terraform/aks/workspace-variables/sandbox.tfvars.json]: Probably not right now, have disabled

File: config/settings/sandbox.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+dfe_sign_in:
+  # URL that the users are redirected to for signing in
+  issuer: https://oidc.signin.education.gov.uk
+  # URL of the users profile
+  profile: https://profile.signin.education.gov.uk
+
+features:
+  basic_auth: true
+  sign_in_method: dfe-sign-in
+  basic_auth: false
```
Comment (d-a-v-e) [inline config/settings/sandbox.yml:16]: this is repeated just above - I assume we want it to false

**Review by d-a-v-e — APPROVED** [review]

File: .github/actions/deploy/action.yml [hunk 5]
```diff
@@ -125,7 +125,7 @@ runs:
 
     - name: Run data migrations
       shell: bash
-      if: inputs.environment == 'staging' || inputs.environment == 'productiondata' || inputs.environment == 'production'
+      if: inputs.environment in ['staging', 'sandbox', 'productiondata', 'production']
```
Comment (saliceti) [inline .github/actions/deploy/action.yml:128]: This syntax doesn't work in github actions: https://github.com/DFE-Digital/register-trainee-teachers/actions/runs/7784096429/job/21224021287?pr=4000
You may want [contains](https://docs.github.com/en/actions/learn-github-actions/expressions#contains) ?

Comment (darokel) [reply .github/actions/deploy/action.yml]: Fixed!

**Review by stevehook — APPROVED** [review]
Can't really comment on the terraform stuff etc. but looks good to me

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**